uid/src/main/java/com/tomtom/kotlin/uid/Uid.kt

```diff
@@ -0,0 +1,217 @@
+        private const val UUID_MAX_LENGTH = 36
+        private val UUID_DASH_POS = intArrayOf(8, 13, 18, 23)
+
+        /**
+         * Returns an ID if it is a valid UUID, or `null` if it's not.
+         *
+         * @param id String representation of UUID.
+         * @return Valid UUID, or null.
+         */
+        fun <T> fromStringIfValid(id: String?) =
+            if (id == null) {
+                null
+            } else try {
+                Uid<T>(id)
+            } catch (ignored: IllegalArgumentException) {
+                null
+            }
+
+        /**
+         * Instantiates a [Uid] with given ID as a string. Mainly used for deserialization.
+         * Opposite of [toString].
+         *
+         * @param <T> Uid type.
+         * @param id  String representation of ID.
+         * @return Uid.
+         * @throws IllegalArgumentException If name does not conform to the string
+         * representation as described in [UUID.toString]. Use [isValid] to make sure the
+         * string is valid.
+        </T> */
+        fun <T> fromString(id: String) = Uid<T>(id)
+
+        /**
+         * Instantiates a [Uid] with given ID as hex-formatted string. Opposite of [toHexString].
+         *
+         * @param <T> Uid type.
+         * @param id  Hex string representation of ID, must be exactly 32 characters long.
+         * @return Uid.
+         * @throws IllegalArgumentException If name does not conform to the string
+         * representation as described in [UUID.toString]. Use [isValid] to make sure the
+         * string is valid.
+        </T> */
+        fun <T> fromHexString(id: String): Uid<T> {
+            require(id.length == 32)
+            val msb: Long =
+                id.substring(0, 8).toLong(16) shl 32 or id.substring(8, 16).toLong(16)
+            val lsb: Long =
+                id.substring(16, 24).toLong(16) shl 32 or id.substring(24, 32).toLong(16)
+            return Uid<T>(UUID(msb, lsb))
+        }
+
+        /**
+         * Return if string contains valid UUID characters only.
+         * Must be converted to lowercase already.
+         *
+         * @param uuid Input UUID. Should be lowercase already.
+         * @return True if valid characters only.
+         */
+        private fun onlyContainsValidUuidCharacters(uuid: String): Boolean {
+            for (ch in uuid.toCharArray()) {
+                if (!(ch in '0'..'9' || ch in 'a'..'f' || ch == UUID_DASH)) {
+                    return false
+                }
+            }
+            return true
+        }
+
+        /**
+         * Checks if the dashes are at the right positions for a UUID.
+         *
+         * @param s Input string.
+         * @return True if the dashes are correctly placed.
+         */
+        private fun areDashesAtCorrectPosition(s: String) =
+            UUID_DASH_POS
+                .map { i -> s.length > i && s[i] == '-' }
+                .reduce { acc, b -> acc && b }
```

Comment:
How about something like this:
```
UUID_DASH_POS.all { s.length > it && s[it] == '-' }
```
Although, I think you could filter out the length check, to check for minimum length of the max of UUID_DASH_POS.

Reply:
OK.